### PR TITLE
Dev3 change

### PR DESCRIPTION
**Pull request by ajphukan**
Changes ..

### CODE REVIEW

File: src/main.c [hunk 1]
```diff
@@ -127,16 +127,15 @@ main (int argc, char *argv[])
 
   }
   while (next_option != -1);
-
-  if (getenv ("HOME") == NULL)
+  
+  char *HOMEDIR = getenv ("HOME");
+  
+  if (HOMEDIR == NULL)
```
Comment (andy5995) [inline src/main.c]: Looks good, but let's "reverse" the condition. If HOMEDIR != NULL do the bufchk else return 1.

Comment (ajphukan) [inline src/main.c]:
```c
if (HOMEDIR != NULL)
{
   if (bufchk (HOMEDIR, MP))
     return BUF_ERR;
}
else
   {
     fprintf (stderr, "Error: Environmental variable $HOME can't be used. Unable to determine home directory\n");
     return 1;
   }
```
you want this ?

Comment (andy5995) [inline src/main.c]: Right, but the else is part of the first if, so the block below it should be indented 2 spaces to the left.

Comment (ajphukan) [inline src/main.c]:
```c
if (HOMEDIR != NULL)
{
   if (bufchk (HOMEDIR, MP))
      return BUF_ERR;
   else
    {
       fprintf (stderr, "Error: Environment variable $HOME can't be used. Unable to determine home directory\n");
       return 1;
    }
}

```
this one i will check the identation in the editor .

Comment (ajphukan) [inline src/main.c]:
```c
if (HOMEDIR != NULL)
{
   if (bufchk (HOMEDIR, MP))
      return BUF_ERR;
   else
    {
       fprintf (stderr, "Error: Environment variable $HOME can't be used. Unable to determine home directory\n");
       return 1;
    }
}

```
this one i will check the identation in the editor .

Comment (andy5995) [inline src/main.c]:
```
if (HOMEDIR != NULL)
{
   if (bufchk (HOMEDIR, MP))
     return BUF_ERR;
}
else
{
   fprintf (stderr, "Error: Environmental variable $HOME can't be used. Unable to determine home directory\n");
   return 1;
}
```

Do you have the IRC program installed?

Comment (ajphukan) [inline src/main.c]: not yet but i will install it soon .. 
```c
if (HOMEDIR == NULL)
  {
    fprintf (stderr, "Error: Environmental variable $HOME can't be used. Unable to determine home directory\n");
    return 1;
  }
```
And we are printing that error when HOMEDIR ==NULL but in above code block it is opposite

Comment (andy5995) [inline src/main.c]: opposite, but same effect. If HOMEDIR != NULL the else will be skipped over. If HOMEDIR == NULL the message will get printed.

**Comment by ajphukan** [comment]
I got confused due to that ... :-P

**Comment by ajphukan** [comment]
Merged  :+1:

**Pull request merged**